pylint/checkers/utils.py

```diff
@@ -1535,3 +1535,28 @@ def get_import_name(
                     modname, level=importnode.level
                 )
     return modname
+
+
+def is_node_in_guarded_import_block(node: astroid.NodeNG) -> bool:
```

Comment:
Would it make sense to have this check directly in ``astroid.If`` as ``is_typing_or_sys_guard`` ? Would permit to refactor to:
```python
return isinstance(node.parent, astroid.If) and node.is_typing_or_sys_guard()
```
Or even use directly without creating a function.

Reply:
See PyCQA/astroid#1099